src/Input/LocalInputSource.php

```diff
@@ -122,16 +129,39 @@ public function isPDFEmpty() // TODO: add PDF lib
     /**
      * Reads the contents of the file.
      *
-     * @param boolean $closeFile Whether to close the file after parsing it.
      * @return array
      */
-    public function readContents(bool $closeFile = true): array
+    public function readContents(): array
     {
         $fileHandle = fopen($this->fileObject->getFilename(), 'rb');
         $strContents = fread($fileHandle, filesize($this->fileObject->getFilename()));
-        if ($closeFile) {
-            fclose($fileHandle);
-        }
+        fclose($fileHandle);
         return [basename($this->fileObject->getFilename()), $strContents];
     }
+
+    /**
+     * Closes the handle/stream, if the input type supports it.
+     *
+     * @return void
+     * @throws MindeeSourceException Throws when strict mode is enabled.
+     */
+    public function close(): void
+    {
+        if ($this->throwsOnClose) {
+            throw new MindeeSourceException("Closing is not implemented on this type of local input source.");
+        } else {
+            error_log("Closing is not implemented on this type of local input source.");
+        }
+    }
```

Comment:
when does this actually do anything?

Reply:
When called on other input sources. It is mostly here for unit tests.